velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#define XXH_INLINE_ALL
+
+#include "velox/functions/prestosql/aggregates/PrestoHasher.h"
+#include "velox/external/xxhash.h"
+
+namespace facebook::velox::aggregate {
+
+namespace {
+
+int64_t rotateLeft(const int64_t num, const int32_t distance) {
+  auto unsignedShift = ((uint64_t)num) >> -distance;
+  return (num << distance) | unsignedShift;
+}
+
+template <typename T>
+FOLLY_ALWAYS_INLINE int64_t getHashForIntTypes(const T& value) {
+  return rotateLeft(value * 0xC2B2AE3D27D4EB4FL, 31) * 0x9E3779B185EBCA87L;
```

Comment:
Where do these constants come from? Would be nice to give them names and add a comment to document their origin.

Reply:
These constants come from Presto. Not sure what to name them, as there arent names there either, but I will try.

Reply:
Do these come from https://github.com/airlift/slice/blob/master/src/main/java/io/airlift/slice/XxHash64.java#L27 ?

I do see them being named PRIME64_1, PRIME64_2,.. PRIME64_5.

Reply:
I also see these constants in velox/external/xxhash.h . Can't we just use some function from there instead of writing our own?